### PR TITLE
sci-libs/tensorflow: fix build with dev-libs/jsoncpp-1.9.2

### PR DESCRIPTION
**Pull request by jok-ts**
Header file name changed in a patch release.

tensorflow-2.0.0-r0 already clamps the dependency to ~dev-libs/jsoncpp-1.9.1, so 
tensorflow-2.0.0-r1 can depend on >=dev-libs/jsoncpp-1.9.2 and avoid conditional patching.

Closes: https://bugs.gentoo.org/701334
Signed-off-by: Jouni Kosonen <jouni.kosonen@tukesoft.com>

### CODE REVIEW

File: sci-libs/tensorflow/files/tensorflow-2.0.0-jsoncpp-1.9.2.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
+diff --git a/third_party/systemlibs/jsoncpp.BUILD b/third_party/systemlibs/jsoncpp.BUILD
+index 7d54f92..ce14265 100644
+--- a/third_party/systemlibs/jsoncpp.BUILD
++++ b/third_party/systemlibs/jsoncpp.BUILD
+@@ -10,7 +10,7 @@ HEADERS = [
+     "include/json/assertions.h",
+     "include/json/autolink.h",
+     "include/json/config.h",
+-    "include/json/features.h",
```
Comment (mgorny) [inline sci-libs/tensorflow/files/tensorflow-2.0.0-jsoncpp-1.9.2.patch]: Do they really need all those headers? `json.h` includes all of them, so this really looks like another clueless upstream.

Comment (jok-ts) [reply sci-libs/tensorflow/files/tensorflow-2.0.0-jsoncpp-1.9.2.patch]: The way they do things, they do. The include/json directory is not added to the includepath, the relevant include files are softlinked to the build tree instead. Softlinking only the json/json.h would not tell the build system where to find the rest of the includes.

This is probably to make the process essentially the same as it is for the bundled jsoncpp, but it is demostrably fragile as it broke with 1.9 and again with 1.9.2.

Comment (mgorny) [reply sci-libs/tensorflow/files/tensorflow-2.0.0-jsoncpp-1.9.2.patch]: /me facepalms to the floor level.

**Comment by jok-ts** [comment]
This has been here a while now, so I think I'll rebase it to a fresh tree and drop the dead python3_5 support in the process.

**Comment by mgorny** [comment]
I'm sorry for failing to reply again earlier. Could you apply the same fix to the two remaining versions, and remove old revisions while at it? This should make it possible to remove old jsoncpp.

**Comment by jok-ts** [comment]
> I'm sorry for failing to reply again earlier. Could you apply the same fix to the two remaining versions, and remove old revisions while at it? This should make it possible to remove old jsoncpp.

All in the same commit? I won't be able to check that the modified ebuilds still work but at least the patch applies.

**Comment by jok-ts** [comment]
FWIW, tensorflow 2.1.0 was released a week ago and it too will need the fix in order to compile against jsoncpp-1.9.2.

**Comment by jok-ts** [comment]
@perfinion Any hope of getting this in the tree? I believe tensorflow is the only package sitting on jsoncpp-1.9.1 now. 

I'd like to clean up my local overlay and try to see why I'm running into [upstream issue 35653](https://github.com/tensorflow/tensorflow/issues/35653) with both 2.1.0_rc2 and 2.1.0 proper.

**Comment by jok-ts** [comment]
Nevermind then, fix seems to be now in the tree with `tensorflow-2.1.0-0002-systemlibs-unbundle-pybind11.patch` addressing the upstream issue too. Closing.

**Pull request closed without merging**